src/frontend/Komponenter/Behandling/Brev/ManueltBrev.tsx

```diff
@@ -0,0 +1,177 @@
+import { useDataHenter } from '../../../App/hooks/felles/useDataHenter';
+import { IManueltBrev, IAvsnitt } from '../../../App/typer/brev';
+
+const StyledManueltBrev = styled.div`
+    width: 50%;
+    margin-bottom: 10rem;
+`;
+
+const Innholdsrad = styled(Panel)`
+    margin-top: 1rem;
+`;
+
+const Knapper = styled.div`
+    margin-top: 2rem;
+
+    display: flex;
+    justify-content: space-between;
+`;
+
+const BrevKolonner = styled.div`
+    display: flex;
+`;
+
+const BrevWrapper = styled.div`
+    margin-left: 4rem;
+`;
+
+interface Props {
+    fagsakId: string;
+}
+
+const ManueltBrev: React.FC<Props> = ({ fagsakId }) => {
+    const førsteRad = [
+        {
+            deloverskrift: '',
+            innhold: '',
+        },
+    ];
+
+    const [overskrift, settOverskrift] = useState('');
+    const [avsnitt, settAvsnitt] = useState<IAvsnitt[]>(førsteRad);
+    const [brev, settBrev] = useState<Ressurs<string>>(byggTomRessurs());
+    const { axiosRequest, innloggetSaksbehandler } = useApp();
+
+    const personopplysningerConfig: AxiosRequestConfig = useMemo(
+        () => ({
+            method: 'GET',
+            url: `/familie-ef-sak/api/personopplysninger/fagsak/${fagsakId}`,
+        }),
+        [fagsakId]
+    );
+
+    const personopplysninger = useDataHenter<IPersonopplysninger, null>(personopplysningerConfig);
+
+    const genererBrev = () => {
+        if (personopplysninger.status !== RessursStatus.SUKSESS) return;
+
+        const brevdato = dagensDatoFormatert();
+
+        axiosRequest<string, IManueltBrev>({
+            method: 'POST',
+            url: `/familie-ef-sak/api/manueltbrev`,
+            data: {
+                overskrift,
+                avsnitt,
+                saksbehandlersignatur: innloggetSaksbehandler?.displayName || 'Ukjent',
+                brevdato,
+                ident: personopplysninger.data.personIdent,
+                navn: personopplysninger.data.navn.visningsnavn,
+            },
+        }).then((respons: Ressurs<string>) => {
+            settBrev(respons);
+        });
+    };
+
+    const leggTilRad = () => {
+        settAvsnitt((eksisterendeAvsnitt: IAvsnitt[]) => {
+            return [
+                ...eksisterendeAvsnitt,
+                {
+                    deloverskrift: '',
+                    innhold: '',
+                },
+            ];
+        });
+    };
+
+    const endreDeloverskrift = (e: React.ChangeEvent<HTMLInputElement>) => {
+        e.preventDefault();
+
+        const index = parseInt(e.target.id, 10);
+
+        settAvsnitt((eksisterendeAvsnitt: IAvsnitt[]) => {
+            const oppdatertAvsnitt = eksisterendeAvsnitt.slice();
+
+            oppdatertAvsnitt[index].deloverskrift = e.target.value;
+
+            return oppdatertAvsnitt;
+        });
+    };
+
+    const endreInnhold = (e: React.ChangeEvent<HTMLInputElement>) => {
+        e.preventDefault();
+
+        const index = parseInt(e.target.id, 10);
```

Comment:
Er det trygt å bruke indeks her eller kan vi få litt trøbbel på sikt? Alternativt kan vi generere en uuid hver gang en rad legges til og bruke den for å finne igjen riktig avsnitt...?

Reply:
Enig, det føles ryddigere, så la på uuid i 48223c1. Byttet også til å bruke data-attributt istedenfor id direkte, da TypeScript likte det bedre.